source/balances/contracts/BalanceChecker.sol

```diff
@@ -0,0 +1,218 @@
+    * @param spenderAddress address
+    * @param tokenAddresses address[]
+    * @return uint256[] token allowances array if possible
+    */
+  function walletAllowances(
+    address userAddress,
+    address spenderAddress,
+    address[] calldata tokenAddresses
+  ) external view returns (uint256[] memory) {
+    require(tokenAddresses.length > 0);
+    uint256[] memory allowances = new uint256[](tokenAddresses.length);
+
+    for(uint256 i = 0; i< tokenAddresses.length; i++) {
+      allowances[i] = tokenAllowance(userAddress, spenderAddress, tokenAddresses[i]);
+    }
+    return allowances;
+  }
+
+  /**
+    * @notice Check the token allowances of multiple wallets for multiple tokens
+    * @dev return array and will fail if large arrays are inputted
+    * @dev Returns array of token allowances in base units
+    * @param userAddresses address[]
+    * @param spenderAddress address
+    * @param tokenAddresses address[]
+    * @return uint256[] token allowances array if possible
+    */
+  function allAllowancesForManyAccounts(
+      address[] calldata userAddresses,
+      address spenderAddress,
+      address[] calldata tokenAddresses
+    ) external view returns (uint256[] memory) {
+    uint256[] memory allowances = new uint256[](
+      tokenAddresses.length * userAddresses.length);
+
+    for(uint256 user = 0; user < userAddresses.length; user++){
+      for(uint256 token = 0; token < tokenAddresses.length; token++) {
+        allowances[(user * tokenAddresses.length) + token] = tokenAllowance(
+          userAddresses[user], spenderAddress, tokenAddresses[token]);
+      }
+    }
+    return allowances;
+  }
+
+  /**
+    * @notice Check the token balances of multiple wallets for multiple tokens
+    * @dev return array and will fail if large arrays are inputted
+    * @dev Returns array of token balances in base units
+    * @param userAddresses address[]
+    * @param tokenAddresses address[]
+    * @return uint256[] token allowances array if possible
+    */
+  function allBalancesForManyAccounts(
+    address[] calldata userAddresses,
+    address[] calldata tokenAddresses
+  ) external view returns (uint256[] memory) {
+    uint256[] memory balances = new uint256[](tokenAddresses.length * userAddresses.length);
+    for(uint256 user = 0; user < userAddresses.length; user++){
+      for(uint256 token = 0; token < tokenAddresses.length; token++){
+        if( tokenAddresses[token] != address(0x0) ) { // ETH address in Etherdelta config
+          balances[(user * tokenAddresses.length) + token] = tokenBalance(userAddresses[user], tokenAddresses[token]);
+        } else {
+          balances[(user * tokenAddresses.length) + token] =  userAddresses[user].balance;
+        }
+      }
+    }
+    return balances;
+  }
+
+  /**
+    * @notice Checks WETH balances
+    * @param wethAddress address WETH addresses in case of several
+    * @param userAddresses address[]
+    * @return uint256[]
+    */
+  function allWETHbalances(
```

Comment:
whats the need for a WETH-specific function? Can this function not be generic for all IERC20s?